pkg/handler/data_recorder_kafka.go

```diff
@@ -143,7 +143,7 @@ func (r *kafkaEvalResult) Length() int {
 
 // Key generates the partition key
 func (r *kafkaEvalResult) Key() string {
-	if r.EvalResult == nil {
+	if r.EvalResult == nil || r.EvalContext == nil {
```

Comment:
What does this change do?

Reply:
an edge case found via the unit test. The line below `r.EvalContext.EntityID` requires that r.EvalContext is not nil